### DISCUSSION TITLE
Welcome to BS-RoFormer Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/BS-RoFormer#18] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lucidrains** [comment]
is speech separation and music separation related fields, or different?

**Reply by ghost** [reply]
I'm really not an expert, just a musician amazed by this domain, so it will not be technical and people with more knowledge will correct me if needed, but to launch the discussions, here I go:

Speech separation is aimed at separating multiple speakers talking at the same time (the "cocktail party" problem) while music separation is aimed at separating multiple instruments playing at the same time, to make it short.
I guess the concepts used are similar, but not aimed at the same goal: identifying different voices (small differences between each voice) VS identifying different instruments (small to huge differences between each instruments). Most of speech related tasks are aimed to mono audio, music related tasks have to handle stereo to be usable in real world situations. Some algos can even take advantage of stereo info to better identify sound sources.

There's currently a hype and search for a holy grail in music source separation user community about multiple vocals parts separation (lead vocals/backing vocals, or more complex, to separate each vocal parts). This is mainly fueled by the singing voice cloning hype, because you need one voice only audio to train a vocal clone model. And this separation task is really close to that "cocktail party" speech separation problem in fact.

Aside from that, I know that some singing voice separation models from [UVR](https://github.com/Anjok07/ultimatevocalremovergui) based on [MDX-net](https://github.com/kuielab/mdx-net) (and trained by [Kimberley Jensen](https://github.com/KimberleyJensen/)) are working surprisingly well for dialog/background-music separation and speech denoising.

I guess some speech-aimed separation algo/AI/concepts could be adapted to music source separation (some were already). There's some really interesting progress in that field too, for example : [Separate And Diffuse: Using a Pretrained Diffusion Model for Improving Source Separation](https://arxiv.org/abs/2301.10752)

**Reply by lucidrains** [reply]
@jarredou wow, didn't expect such a lengthy response! this was informative, thank you!

**Reply by faroit** [reply]
@lucidrains @jarredou i could probably answer even more lengthy but I just wanted to drop here that we are organizing a [source separation workshop](https://sdx-workshop.github.io/) next week (registration is free) and the authors of the [RoFormer will be there to present](https://github.com/sdx-workshop/sdx-submissions/pull/4) the model.

**Reply by lucidrains** [reply]
@faroit awesome! won't be able to make it, but definitely want to return to Italy at some point (was at Lake Como for a wedding earlier this year)

**Reply by ghost** [reply]
I wish I could free myself for it. I see that [ZFTurbo](https://github.com/sdx-workshop/sdx-submissions/pull/9) is also participating to the workshop ! Nice ! :)

**Reply by turian** [reply]
@lucidrains I like @jareddou's analysis, in particular pointing to the cocktail party problem.

Historically, speech and music ML have been very different, but I think they are converging because of increasingly sophisticated ML algorithms. (With that said, speech or music people will typically choose hyperparameters and do model design based upon their domain, and then sometimes miss certain error classes of other domains that still require some expert knowledge. I think this is particularly acute in audio ML versus NLP or vision ML because our evaluation measures like SDR etc aren't that close to human perception, which is poorly understand.)

But to return to the key question about MSS versus speech separation, the difference is one of problem formulation. Separating speech from non-speech is relatively simply formulated, similar to vocal and instrumental separation. Separating individual speakers is a much more challenging formulation, because you're essentially predicting an ordered set.

MSS thought of as 4 (or 6 well specified instruments) makes it easy to formulate, but a bit boring. I think a lot of music producers would be very interested in separating music into all the individual instruments. This wouldn't really be done by having 100 different instrument models (and that wouldn't span all identifiable instruments anyway), rather by *thinking of music source separation as predicting an unordered set of stems*.

**Comment by ghost** [comment]
A new potential SOTA in the domain was published few days ago, less ressources hungry than BS/MelRoformer it seems.
It would be great to see it open-sourced if authors don't publish their code !
https://arxiv.org/abs/2401.13276

**Comment by ghost** [comment]
A new interesting source separation paper from ByteDance
![image](https://github.com/user-attachments/assets/98b9a84a-35dd-4304-83c8-803ede7a1786)

https://arxiv.org/abs/2506.16833
